python/xir/program.py

```diff
@@ -258,167 +260,252 @@ class XIRProgram:
+        self._gates = {}
+        self._operators = {}
 
-        self._called_ops = set()
+        self._variables = set()
+        self._called_functions = set()
 
     def __repr__(self) -> str:
+        """Returns a string representation of the XIR program."""
         return f"<XIRProgram: version={self._version}>"
 
     @property
     def version(self) -> str:
-        """Version number of the program
+        """Returns the version number of the XIR program.
 
         Returns:
-            str: program version number
+            str: version number
         """
         return self._version
 
     @property
     def use_floats(self) -> bool:
+        """Returns whether floats and complex types are used by the XIR program.
+
+        Returns:
+            bool: whether floats and complex types are used
+        """
         return self._use_floats
 
     @property
-    def wires(self) -> Set[int]:
-        """Get the wires of an XIR program"""
+    def wires(self) -> Collection[Wire]:
+        """Returns the wires of the XIR program.
+
+        Returns:
+            Collection[Wire]: collection of wires
+        """
         wires = []
         for stmt in self.statements:
             wires.extend(stmt.wires)
 
         return set(wires)
 
     @property
-    def include(self) -> List[str]:
-        """Included XIR libraries/files used in the program
+    def called_functions(self) -> Collection[str]:
+        """Returns the functions that are called in the XIR program.
 
         Returns:
-            list[str]: included libraries/files
+            Collection[str]: collection of function names
         """
-        return self._include
+        return self._called_functions
 
     @property
-    def options(self) -> Dict[str, Any]:
-        """Script-level options declared in the program
+    def declarations(self) -> Mapping[str, Sequence[Declaration]]:
+        """Returns the declarations in the XIR program.
 
         Returns:
-            Dict: declared scipt-level options
+            Mapping[str, Sequence[Declaration]]: dictionary of declarations
+                sorted into the following keys: 'gate', 'func', 'output' and
+                'operator'.
         """
-        if self.use_floats:
-            options_with_floats = get_floats(self._options)
-            if isinstance(options_with_floats, Dict):
-                return options_with_floats
-        return self._options
+        return self._declarations
 
     @property
-    def statements(self) -> List[Statement]:
-        """Statements in the program
+    def gates(self) -> Mapping[str, Mapping[str, Sequence]]:
+        """Returns the gates in the XIR program.
 
         Returns:
-            list[Statement]: a list of all statements
+            Mapping[str, Mapping[str, Sequence]]: dictionary of gates, each gate
+                consisting of a name and a dictionary with the following keys:
+                'parameters', 'wires' and 'statements'
```

Comment:
Same here as before.
```suggestion
            Mapping[str, Mapping[str, Sequence]]: dictionary of gates, each gate
            consisting of a name and a dictionary with the following keys:
            'parameters', 'wires' and 'statements'
```